packages/govuk-frontend/src/govuk/govuk-frontend-component.mjs

```diff
@@ -14,9 +14,29 @@ export class GOVUKFrontendComponent {
    * Constructs a new component, validating that GOV.UK Frontend is supported
    *
    * @internal
+   * @param {Element | null} [$module] - HTML element to use for component
+   * @param {string | null} [moduleName] - name of module
    */
-  constructor() {
+  constructor($module, moduleName) {
```

Comment:
**suggestion** The `moduleName` set on a child class can be accessed via `this.constructor.moduleName`. This would avoid classes inheriting to have to pass the `moduleName` to the constructor.

TypeScript may have complaints about this, as there's nothing in the types saying that the child class would have that static property. The base class could check for something like `if (typeof this.constructor.moduleName !== 'string')` which would both set the type and let users know they're inheriting the class properly 🤔

Reply:
If we added that check in the constructor, we'd then be able to make the `InitError` more informative:
- say that the component is already initialised if `moduleName` is a string
- say that the `moduleName` is not the right type if it's anything else

Reply:
I'm still wondering if we shouldn't have `moduleName` be a mandatory thing to specify. Since it is used in other places also to create and init components.